### PR TITLE
linux-4.20: fix cross compile for linux-headers package

### PR DESCRIPTION
**Pull request by marmeladema**

### CODE REVIEW

**Comment by marmeladema** [comment]
This will help fixing #4641

**Comment by Hoshpak** [comment]
Since the changes are non-trivial, it will take some time to review and test. If you don't hear from me within a few days, feel free to remind me.

**Comment by marmeladema** [comment]
Basically it adds more file (Makefile, Kconfig, Kbuild, etc) in the linux-headers package for every architectures. Debian does the same thing.
It removes host-built helper binaries from the linux-headers package when cross compiling.
And add some missing directories needed to rebuilt those helper binaries on the target arch.

File: srcpkgs/linux4.20/template [hunk 1]
```diff
@@ -140,6 +140,13 @@ do_install() {
 			;;
 	esac
 
+	if [ "$CROSS_BUILD" ]; then
+		# Will transform x86_64 to x84-64 as reported by 'file' command
+		machine=$(echo $XBPS_MACHINE | tr '_' '-')
```
Comment (maxice8) [inline srcpkgs/linux4.20/template]: can be 

```
machine=${XBPS_MACHINE//_/-}
```

Comment (maxice8) [reply srcpkgs/linux4.20/template]: Also, you declared this variable to use it only once ?

**Comment by Hoshpak** [comment]
I successfully compiled a simple module with dkms on x86_64 with these changes so there seem to be no regressions. I completely agree with removing the x86_64 binaries, it just makes no sense to ship them on foreign architectures. However I'm very sceptical regarding the addition of 2406 Makefiles. Can't this be narrowed down to just the ones needed to achieve the result you want?

**Comment by marmeladema** [comment]
@Hoshpak i squashed everything in one commit and fixed @maxice8 comment. I have also reverted my change to keep `arch/x86` to keep only `arch/x86/ras/Kconfig` instead.

**Comment by marmeladema** [comment]
I confirm i successfully build an external module on an aarch64 board.

**Pull request merged**